src/python/pants/option/arg_splitter.py

```diff
@@ -79,7 +79,12 @@ class ArgSplitter:
 
     def __init__(self, known_scope_infos: Iterable[ScopeInfo], buildroot: str) -> None:
         self._known_goal_scopes = dict(self._get_known_goal_scopes(known_scope_infos))
-        self._native_arg_splitter = PyArgSplitter(buildroot, list(self._known_goal_scopes.keys()))
+        native_known_goals = tuple(
```

Comment:
Note that we can't derive this from `self._known_goal_scopes` because that has already flattened out the aliases, so we recompute it from the underlying data.

A followup change will get rid of `self._known_goal_scopes` anyway. In fact, it will probably get rid of this entire class...